### DISCUSSION TITLE
Add Support for old Apple Cinema Displays (USB Brightness Control)

### DISCUSSION THREAD

**Discussion by biho1** [MonitorControl/MonitorControl#891] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Is your feature request related to a problem? Please describe

Apple Cinema Display 30" with Club3D USB-C to DualLink DVI Adapter on M1 Mac Mini doesn't responde to brightness changes. 

Also Tested the original DualLink DVI to Mini DisplayPort Adapter with some Displayport to USB-C Adapters and also no brightness change.

All tested with Bug Sur and Monterey.

Seems like the DDC commands don't reacht the old ACD30, it should support them.

### Describe the solution you'd like

The old Apple Cinema Display (20,23,30 and LED 24,27) are also connected with USB 2.0 and supports brightness control over USB.

With added implementation of USB brightness control for the old Apple Cinema Displays the would be a fallback if DDC is not working.

a working solution for linux in C++ can be found here https://github.com/warvariuc/acdcontrol (is tested it with Ubuntu 20)

supported devices are listed in acdcontrol.cpp, also the USB device IDs are listed 

### Describe alternatives you've considered

fix DCC communication for the old Cinema Displays and ensure pass through over whatever adapter used

**Comment by waydabber** [comment]
See this as well: https://github.com/MonitorControl/MonitorControl/issues/364

If somebody is willing, can contribute by adding USB support for the affected displays. I don't think the core dev team will move into this direction anytime soon though to be honest.

**Reply by kstella** [reply]
my old Apple LED Cinema Display does with this, even now that I'm on M2/Ventura (lost some fine controls though). 
LED is connect to MB via USBCele dongle. Not all dongles I tried worked. 
The only only odd thing, each time I log in, it is now set to full brightness, while the Thunderbolt Display retains setting.

**Comment by newkind** [comment]
I'm using two Apple Cinema Displays 30" with Choetech USB-C -> Mini DisplayPort adapters and I can tell you that the brightness controls (via keyboard) are working, but not 100% properly. I cannot go as bright as the monitor allows to. When I set the brightness to the max via keyboard its still dark and I have to use the buttons on the side of the monitor to get it even brighter. Besides of that - increasing and decreasing brightness works for me.

**Comment by waydabber** [comment]
@newkind You might have to change the DDC max setting to 255 for an extended range.

<img width="798" alt="Screen Shot 2022-01-10 at 11 50 50" src="https://user-images.githubusercontent.com/37590873/148754061-2ad64f74-82ec-4a6e-9572-77cc1081aa37.png">

**Reply by thunsteath** [reply]
Hi I'm using the Cinema Display 30" with a Mac Mini M 2 but I can't  set the control method to  Hardware (DDC) so no way to fill in the DDC max with 255 value. What do I do to switch from Software (gamma) to Hardware (DDC)? 
![Screenshot 2023-03-04 at 13 00 03](https://user-images.githubusercontent.com/126959267/222900126-9b989e0d-66a8-4fa7-84ce-981d9cf14494.png)

**Reply by waydabber** [reply]
How is the display connected?

**Comment by newkind** [comment]
@waydabber that worked perfectly! thank you!

**Reply by thunsteath** [reply]
Hi I'm using the Cinema Display 30" with a Mac Mini M 2 but I can't  set the control method to  Hardware (DDC) so no way to fill in the DDC max with 255 value. What do I do to switch from Software (gamma) to Hardware (DDC)? 
![Screenshot 2023-03-04 at 13 00 03](https://user-images.githubusercontent.com/126959267/222900108-8950fae2-5390-495d-9fbf-cb5452d47487.png)

**Comment by waydabber** [comment]
Thanks for the response. I'll convert this into discussion for future edification.

**Reply by evanthorpe** [reply]
Hi! The max DDC values don’t seem to be changing anything in my 27” Apple Cinema Display? I have it connected via an adapter to an Anker dock and then via USB C to my 14” M1 Pro. The Apple Cinema Display seems a bit dim compared to what it should be able to do, but setting the DDC max values in Advanced Settings doesn’t seem to make any difference. Any ideas?

**Comment by thunsteath** [comment]
With the standard cable it came with, connected to the Club 3D dual DVI tio usb cable. The Cinema displays usb-cable is connected to the Mini M2 Pro en the Club 3D I’ve tried on al 4 Thunderbolt 4 ports.
I used to connect the Cinema display to a genuine Apple DVI to Thunderbolt2 adapter and connected it to a 2017 Mac Mini via Thunderbolt2 and everything was fine, maximum brightness, no control possible via the keyboard or screen buttons but all done by the system (Big Sur) with the built in light sensor.

Nog with the M2 Pro control via keyboard and screen buttons is possible but I can't get full brightness, it’s allways a bit yellowish.
So I hopen, with the good reviews and experiences, that MonitorControl would solvemy issue, but unfortunately…unless I do something right, or my settings (like Hardware (DDC) iso Software (Gamma) could be. the reason. But no way to change that setting.

Thanks in advance for looping into it!

regzards,
Valentijn

**Reply by waydabber** [reply]
Hmm. It's kinda exotic. :) But I don't know what could be the issue - from the looks of it it's like the Mac detects the display as a virtual display. You can install [BetterDisplay](https://github.com/waydabber/BetterDisplay#readme) and create a screenshot of Settings/Displays with the info section(s) open so I might see more about the display.

**Comment by thunsteath** [comment]
Hi,

I’ve installed Better Display, but for me and my limited knowledge of the matter it’s getting a bit an overload of settings just to get my brightness right :-)
I hope I took the right screenshots (see below). I must also say the in the beginning I’ve experimented with attaching my iPad Air 4th gen as a secondary display, but disconnected it after maybe 15minutes. Don’t know if the Mini might remember that ’secondary display’ as a virtual display, or kept memory aside for this construction.
As I said I’m very much into using the mac but not so much in all the technicality behind, more the plug and play type counting on SJ’s saying ‘It just works!
If I win the lotery on wednesday I buy myself a nice Mac studio with the upcoming XDR refresh :-)
￼
![Screenshot 2023-03-06 at 11 57 47](https://user-images.githubusercontent.com/126959267/223113986-0bb9c655-8ed7-4fb0-b011-26633a047c59.png)
![Screenshot 2023-03-06 at 11 55 43](https://user-images.githubusercontent.com/126959267/223114033-cd021756-793e-4c97-ab53-9870d2f1129f.png)
![Screenshot 2023-03-06 at 11 56 14](https://user-images.githubusercontent.com/126959267/223114074-5ce13b6a-dddd-450f-b9a0-46ec559de5e6.png)

￼
￼Maybe the 'Obsoleted Apple Display' says it all...

**Reply by waydabber** [reply]
The screenshots did not come through, you might have to post it on GitHub, email attachments don't work.

**Reply by waydabber** [reply]
All right, the app properly detects the display as an obsoleted Apple device. The app will not enable DDC for these displays by default and Apple control does not work as well. If the 30" Cinema Display is DDC compatible, you can try enabling this (hopefully this option should show up, never actually tested on a display like this):

<img width="942" alt="Screenshot 2023-03-06 at 14 14 21" src="https://user-images.githubusercontent.com/37590873/223120445-df1d6eb1-9f37-4c0d-8e59-3d601dd30e8f.png">

Let me know if it does not show up.

**Reply by thunsteath** [reply]
Hey,

Thanks a lot, but I couldn’t enable DDC. I can treat it as generic third-party (non Apple) display but still no way in Monitor Control to change the Control methode to DDC. I’ve just tried all the same with a Cinema Display 23”, but the results are the same, less brightness an no way to control via DDC. I’ve read somewhere that the problem might be Ventura, it doesn’t support the Cinema Display drivers anymore or so. People with the M1 their machine came with Monterey and that one still does support them but mine came with Ventura so no way to check. (and it just a read no idea if it’s true).
Anyway good to know what’s all possible! ...and everything works with my 2005 display and 2023 Mini except for the highest brightness so...

Now the real dilemma: sticking with Apple and going for an overpriced, low specs 27" studio display or for that amount, 3 Samsung M8 32” displays ;-)

Thanks a lot for all the help and quick responses!

Ciao

**Reply by waydabber** [reply]
Yes, if you change that in BetterDisplay, it won't affect MonitorControl. You should try using the brightness slider of BD. In MC you can't fix this and support probably won't be added unless somebody posts a PR regarding this.

**Comment by thunsteath** [comment]
sorry, updated and editen the post on GitHub
